daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/Facets.scala

```diff
@@ -387,7 +413,7 @@ trait Facets { self: Restriction =>
         errorOnLocalLessThanBaseFacet(theLocalFacet, theRemoteFacet, facetType)
         localFacet
       }
-      case Facet.maxLength | Facet.fractionDigits => {
+      case Facet.length | Facet.maxLength | Facet.fractionDigits => {
         errorOnLocalGreaterThanBaseFacet(theLocalFacet, theRemoteFacet, facetType)
```

Comment:
This means that a local length facet must be less than the base length facet, is that correct? Have we confirmed with the XSD spec that that is the right behavior? Feels like make a local length should always be the same as a base length, since there is no way to narrow a single value instead of a range like min/maxValue have.

Reply:
Found this in the XSD spec:
> Schema Component Constraint: length valid restriction
It is an [·error·](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#dt-error) if [length](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#f-l) is among the members of [{facets}](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#std-facets) of [{base type definition}](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#std-base_type_definition) and [{value}](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#f-l-value) is not equal to the [{value}](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#f-l-value) of the parent [length](https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html#f-l).

So I think this is wrong. This needs to error if local is not equal to base.